### PR TITLE
Some cleanup and Updates to vds loading to support sample filtering

### PR DESCRIPTION
**Pull request by jkgoodrich**

### CODE REVIEW

File: gnomad_qc/v4/create_release/create_de_novo_release.py [hunk 1]
```diff
@@ -223,13 +241,114 @@ def custom_an_select_globals(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
     }
 
 
+def get_de_novo_config() -> Dict[str, dict]:
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_de_novo_release.py:244]: We'll leave this as it is now?

**Review by KoalaQin — COMMENTED** [review]
I don't have many questions about the dense mt part, but I didn't look at the other parts carefully, the test ran successfully.

File: gnomad_qc/v4/create_release/create_de_novo_release.py [hunk 2]
```diff
@@ -95,47 +104,56 @@ def filter_de_novos(ht: hl.Table) -> hl.Table:
     return ht
 
 
-def densify_releasable_trios_mt(
-    vds: hl.vds.VariantDataset, fam_ht: hl.Table, var_ht: hl.Table
+def get_dense_releasable_trios_mt(
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v4/create_release/create_de_novo_release.py:107]: Sounds good, can you make the changes?

File: gnomad_qc/v4/create_release/create_de_novo_release.py (hunk 1 above)
Comment (jkgoodrich) [reply gnomad_qc/v4/create_release/create_de_novo_release.py:244]: I'm not sure exactly what the question is

File: gnomad_qc/v4/resources/basics.py [hunk 3]
```diff
@@ -299,10 +313,20 @@ def _remove_ukb_dup_by_index(
                     " filtering) only..."
                 )
                 filter_ht = hard_filtered_samples.ht()
-            else:
+            elif remove_hard_filtered_samples_no_sex:
                 logger.info("Filtering VDS to hard filtered samples only...")
                 filter_ht = hard_filtered_samples_no_sex.ht()
 
+            if filter_samples_ht is not None:
+                logger.info("Filtering VDS to samples in provided Table...")
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v4/resources/basics.py:321]: Sounds good, would you mind adding?

[the remark this answers is not in the document]
Comment (KoalaQin) [reply gnomad_qc/v4/resources/basics.py:321]: dd

Comment (KoalaQin) [inline gnomad_qc/v4/resources/basics.py:321]:
```suggestion
```

File: gnomad_qc/v4/resources/basics.py [hunk 4]
```diff
@@ -299,10 +313,20 @@ def _remove_ukb_dup_by_index(
...
+                if filter_ht is None:
+                    filter_ht = filter_samples_ht
+                elif keep_samples:
+                    filter_ht = filter_samples_ht.semi_join(filter_ht)
+                else:
+                    filter_ht = filter_samples_ht.anti_join(filter_ht)
```
Comment (KoalaQin) [inline gnomad_qc/v4/resources/basics.py:327]:
```suggestion
                    filter_ht = filter_samples_ht.anti_join(filter_ht)
                logger.info("Filtering VDS to %d samples in provided Table...", filter_ht.count()) 
```

File: gnomad_qc/v4/create_release/create_de_novo_release.py (hunk 2 above)
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_de_novo_release.py:107]:
```suggestion
def get_releasable_trios_dense_mt(
```

File: gnomad_qc/v4/create_release/create_de_novo_release.py (hunk 1 above)
Comment (KoalaQin) [reply gnomad_qc/v4/create_release/create_de_novo_release.py:244]: Oh, I meant I just tested it and haven't looked at it carefully.

**Review by KoalaQin — CHANGES_REQUESTED** [review]

File: gnomad_qc/v4/create_release/create_de_novo_release.py (hunk 1 above)
Comment (jkgoodrich) [reply gnomad_qc/v4/create_release/create_de_novo_release.py:244]: Yeah that's fine, it will probably change, but I think we can put it in as is for now

**Review by KoalaQin — APPROVED** [review]
LGTM!

**Pull request merged**